controller/src/main/java/io/pravega/controller/store/kvtable/InMemoryKVTMetadataStore.java

```diff
@@ -57,20 +51,12 @@ KeyValueTable newKeyValueTable(String scope, String kvTableName) {
 
     @Override
     public KeyValueTable getKVTable(String scope, final String name, KVTOperationContext context) {
-        KeyValueTable kvt;
-        if (context != null) {
-            kvt = context.getKvTable();
-            assert kvt.getScopeName().equals(scope);
-            assert kvt.getName().equals(name);
-        } else {
-            if (!scopes.containsKey(scope)) {
-                return new InMemoryKVTable(scope, name);
-            }
-            InMemoryScope kvtScope = scopes.get(scope);
-            Optional<InMemoryKVTable> kvTable = kvtScope.getKVTableFromScope(name);
-            kvt = kvTable.orElse(new InMemoryKVTable(scope, name));
+        if (!scopes.containsKey(scope)) {
```

Comment:
why is the check on scopes? 
shouldnt the check be on table.. if the table doesnt exist, create it.. else return an already created table.. and this should be concurrency protected.

Reply:
and why are we creating kvt object if the scope doesnt exist?
this logic seems incorrect.

Reply:
This was added because scopes created in StreamStore are not available inside KVTStore. Will fix this as part of fix for issue #4960.

Reply:
still, the condition being added here with scopes doesnt exist seems to be introducing an incorrectness. Perhaps we should fix 4960 first before we fix this.

Reply:
Sure. I can raise the PR for 4960 and merge this after that.

Reply:
Fixed as part of this PR.